src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -224,34 +277,41 @@ public function process_directives( string $html ): string {
 			return $html;
 		}
 
-		$context_stack   = array();
-		$namespace_stack = array();
-		$result          = $this->process_directives_args( $html, $context_stack, $namespace_stack );
+		$result = $this->process_directives_args( $html );
+
 		return null === $result ? $html : $result;
 	}
 
 	/**
 	 * Processes the interactivity directives contained within the HTML content
 	 * and updates the markup accordingly.
 	 *
-	 * It needs the context and namespace stacks to be passed by reference, and
-	 * it returns null if the HTML contains unbalanced tags.
+	 * It uses the WP_Interactivity_API instance's context and namespace stacks,
+	 * which are shared between all calls.
+	 *
+	 * This method returns null if the HTML contains unbalanced tags.
 	 *
 	 * @since 6.5.0
+	 * @since 6.6.0 Removed `namespace_stack` and `context_stack` arguments.
 	 *
-	 * @param string $html            The HTML content to process.
-	 * @param array  $context_stack   The reference to the array used to keep track of contexts during processing.
-	 * @param array  $namespace_stack The reference to the array used to manage namespaces during processing.
+	 * @param string $html The HTML content to process.
 	 * @return string|null The processed HTML content. It returns null when the HTML contains unbalanced tags.
 	 */
-	private function process_directives_args( string $html, array &$context_stack, array &$namespace_stack ) {
+	private function process_directives_args( string $html ) {
```

Comment:
Why is the name wrong now?

Reply:
Not that it's _wrong_ now, haha. But maybe it is? 😄 I just thought that the `_args` part in the name referred to the context and namespace stacks being passed as _args_, so it was no longer relevant.

Reply:
Oh, I see. So theres `process_directives` (public function) which calls `process_directives_args`, and that suffix doesn't make sense.

Yes, I think it could be updated, it's private anyways. Maybe it could just be `_process_directives`?

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/6394/commits/2215c9af60fa4c76eed8aadabb6074b8cd63146a.

PS: I tried to come up with a more descriptive name (with no success). 😅